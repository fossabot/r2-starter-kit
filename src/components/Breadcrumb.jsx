import React, { PropTypes } from 'react';
import { Link } from 'react-router-dom';

function Breadcrumb({ data }) {
    return (
        <div className="ts breadcrumb">
            {data.map((val, key) => {
                const lastOne = key === data.length - 1;
                return (
                    <div key={`crumb${key}`} className={`${lastOne ? 'active' : ''} section`}>
                        {val.link
                            ? <Link to={val.link}>{val.text}</Link>
                            : val.text
                        }
                        {!lastOne && <i className="right angle icon divider"></i>}
                    </div>
                );
            })}
        </div>
    );
}

Breadcrumb.propTypes = {
    data: PropTypes.arrayOf(PropTypes.shape({
        link: PropTypes.string,
        text: PropTypes.string
    }))
};

export default Breadcrumb;
