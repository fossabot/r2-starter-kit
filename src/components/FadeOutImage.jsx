/* global setTimeout */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

export default class FadeOutImage extends Component {
  static propTypes = {
    url: PropTypes.string.isRequired,
    placeholder: PropTypes.string.isRequired,
  };
  constructor(props) {
    super();
    this.state = {
      url: props.placeholder,
      blur: 5,
    };
  }
  handleImageOnload = (url) => {
    this.setState({ url, blur: 0 });
  };
  render() {
    const { url, blur } = this.state;
    const { url: selfUrl } = this.props;
    return (
      <div>
        <Image ref={(c) => (this.component = c)} url={url} blur={blur} />
        <img
          id="preload_image"
          src={selfUrl}
          onLoad={() => setTimeout(() => this.handleImageOnload(selfUrl), 500)}
          style={{ display: 'none' }}
        />
      </div>
    );
  }
}

const Image = styled.img`
  display: block;
  width: 100%;
  padding-bottom: 56.25%;
  background: ${({ url }) => `url('${url}') no-repeat center center`};
  background-size: cover;
  filter: ${({ blur }) => `blur(${blur}px)`};
  transition: 0.5s;
`;
