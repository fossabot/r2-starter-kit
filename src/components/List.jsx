import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export default function List({ data }) {
    return (
        <div className="ts divided items">
            { data.map((val, key) => {
                return (
                    <div key={`item${key}`} className="item">
                        <div className="image">
                            <img src={val.img} />
                        </div>
                        <div className="content">
                            <Link to={val.link} className="header">{val.title}</Link>
                            <div className="description">
                                {val.desc}
                            </div>
                        </div>
                    </div>
                );
            })}
        </div>
    );
}

List.defaultProps = {
    data: [{
        id: 0,
        title: 'Title',
        link: '#',
        desc: 'Content',
        img: 'http://placehold.it/320x180',
    }]
};

List.propTypes = {
    data: PropTypes.arrayOf(PropTypes.shape({
        img: PropTypes.string,
        link: PropTypes.string,
        title: PropTypes.string,
        desc: PropTypes.string
    }))
};
