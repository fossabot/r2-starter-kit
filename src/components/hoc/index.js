export { default as withLifecycle } from './WithLifecycle';
