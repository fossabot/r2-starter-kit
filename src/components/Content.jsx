import React, { Component } from 'react';
import PropTypes from 'prop-types';

import FadeOutImage from 'components/FadeOutImage';

export default class Content extends Component {
    render() {
        const {
            id,
            title,
            author = {
                name: '',
                email: ''
            },
            body
        } = this.props;
        return (
            <div>
                {id &&
                    <div className="ts card text container">
                        <div className="image">
                            <FadeOutImage
                              url={`https://picsum.photos/1200/720?image=${id}`}
                              placeholderUrl={`https://picsum.photos/60/36?image=${id}`}
                            />
                        </div>
                        <div className="content">
                            <div className="header">
                                {title}
                            </div>
                            <div className="bulleted meta">
                                {`${author.name} (${author.email})`}
                            </div>
                            <div className="description" dangerouslySetInnerHTML={{ __html: body }} />
                        </div>
                    </div>
                }
            </div>
        );
    }
    static propTypes = {
        id: PropTypes.number,
        title: PropTypes.string,
        author: PropTypes.shape({
            name: PropTypes.string,
            email: PropTypes.string
        }),
        body: PropTypes.string
    }
}
