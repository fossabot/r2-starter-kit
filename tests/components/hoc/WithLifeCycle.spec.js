import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { withLifecycle } from 'components/hoc';

describe('withLifecycle', () => {
  let wrapper;
  let handleDidMount;
  let handleWillUnmount;

  beforeEach(() => {
    handleDidMount = jest.fn();
    handleWillUnmount = jest.fn();
    const WrappedComponent = withLifecycle()(() => <h1>hello</h1>);
    wrapper = shallow(
      <WrappedComponent
        handleDidMount={handleDidMount}
        handleWillUnmount={handleWillUnmount}
      />
    );
  });

  it('should render with initial props', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render with undefined props', () => {
    wrapper.setProps({
      handleDidMount: undefined,
      handleWillUnmount: undefined,
    });
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('handleDidMount is called', () => {
    expect(handleDidMount).toBeCalled();
  });

  it('handleWillUnmount is called', () => {
    wrapper.unmount();
    expect(handleWillUnmount).toBeCalled();
  });
});
