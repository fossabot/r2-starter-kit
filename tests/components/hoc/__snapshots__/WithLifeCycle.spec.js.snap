// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLifecycle should render with initial props 1`] = `<Component />`;

exports[`withLifecycle should render with undefined props 1`] = `<Component />`;
